cpp/src/cylon/net/ucc/ucc_operations.cpp

```diff
@@ -313,6 +333,19 @@ Status UccTableGatherImpl::IgatherBufferData(
         (ucc_aint_t *)displacements.data();
     args.dst.info_v.datatype = UCC_DT_UINT8;
     args.dst.info_v.mem_type = UCC_MEMORY_TYPE_HOST;
+  } else {
+    int sum = 0;
+    auto& recv_counts_ = (*all_recv_counts_)[buf_idx];
+    for(auto count: recv_counts_) {
+      sum += count;
+    }
+    recv_data_placeholder = new uint8_t[sum];
+    args.dst.info_v.buffer = recv_data_placeholder;
+
+    args.dst.info_v.counts = (ucc_count_t *)(*all_recv_counts_)[buf_idx].data();
+    args.dst.info_v.displacements = (ucc_aint_t *)(*displacements_)[buf_idx].data();
+    args.dst.info_v.datatype = UCC_DT_UINT8;
+    args.dst.info_v.mem_type = UCC_MEMORY_TYPE_HOST;
```

Comment:
I actually think we dont need these! we just need to pass `recv_data_placeholder` buffer and make sure everyone (other than the root) copies data to this dummy buffer. So you might not even need to track `all_recv_counts_` and `displacements_`.